FastSim BugFix : PreMix : fix segfault, fix L1 reco, fix digisim link for muons, fix track validation

lveldere opened the pull request:
Lots of fixes in one pr, but they are all entangled
### Fix L1 reco, fix seg fault in premix relvals

FastSim no longer runs
SLHCUpgradeSimulations/Configuration/postLS1CustomsPreMixing.customisePostLS1
because it overwrites modification defined in the fastsim era and the run2 eras

New EDAlias for caloStage1LegacyFormatDigis to make the l1extraParticles function properly.

Note: FullSim still runs SLHCUpgradeSimulations/Configuration/postLS1CustomsPreMixing.customisePostLS1 in the DIGI step.
At some point also FullSim should move away from this.
### Fix digi-sim links for muons

FastSim now runs
SimGeneral/DataMixingModule/customiseForPremixingInput.customiseForPreMixingInput
This customisation function should not affect anything but validation and dqm,
and the customisation function was modified accordingly.
Note: FullSim had never trouble with this because there the customisation function is only applied on the RECO,VALIDATION,DQM jobs.
### Fix track validation

Thanks to SimGeneral/DataMixingModule/customiseForPremixingInput.customiseForPreMixingInput, the tracking particle collection should now include PU tracks

kpedro88 committed:
use QIEType instead of old transient qieIndex

Commit:
Removes subsequent unused return statement in createIfNeeded

Removes subsequent unused return statement and resolves 3742 warnings
from Clang (3.7.1).

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Commit:
Utilities/XrdAdaptor: fix -Wunused-result errors

Some xrootd methods are marked with `XRD_WARN_UNUSED_RESULT`
(`__attribute__((warn_unused_result))`) and `-Wunused-result` forces us
to not ignore return values.

    src/Utilities/XrdAdaptor/src/XrdSource.cc:327:9: error: ignoring return value of function declared with warn_unused_result attribute [-Werror,-Wunused-result]
    src/Utilities/XrdAdaptor/src/XrdSource.cc:338:9: error: ignoring return value of function declared with warn_unused_result attribute [-Werror,-Wunused-result]
    src/Utilities/XrdAdaptor/src/XrdRequestManager.cc:102:9: error: ignoring return value of function declared with warn_unused_result attribute [-Werror,-Wunused-result]
    src/Utilities/XrdAdaptor/plugins/XrdStorageMaker.cc:86:5: error: ignoring return value of function declared with warn_unused_result attribute [-Werror,-Wunused-result]

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Martin-Grunewald committed:
MTport

Commit:
don't run conversion track validation for fastsim

civanch committed:
fixed warnings rported by the static analyzer

Commit:
FastSim BugFix : PreMix : fix L1 reco, fix digisim link in muon digis

lveldere commented:
please test

cmsbuild commented:
A new Pull Request was created by  @lveldere  for CMSSW_8_0_X.

It involves the following packages:

Configuration/PyReleaseValidation
Configuration/StandardSequences
FastSimulation/Configuration
SimGeneral/DataMixingModule


@civanch, @lveldere, @boudoul, @franzoni, @mdhildreth, @fabozzi, @cmsbuild, @ssekmen, @srimanob, @hengne, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @wmtan, @makortel, @GiacomoSguazzoni, @rovere, @VinInn, @Martin-Grunewald, @matt-komm, @cerati, @dgulhan this is something you requested to watch as well.
@slava77, @degano, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

cmsbuild commented:
The tests are being triggered in jenkins.

Commit:
fastsim : exit cmsDriver when using option --fast with fastsim-incompatible scenarios

cmsbuild commented:
-1 
Tested at: f6e92960c370c908363cbd22d01c15063741f467
 When I ran the RelVals I found an error in the following worklfows: 
 4.22 step2
<pre>runTheMatrix-results/4.22_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC/step2_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC.log

----- Begin Fatal Exception 26-Jan-2016 15:02:04 CET-----------------------

An exception of category 'FallbackFileOpenError' occurred while

   [0] Constructing the EventProcessor

   [1] Constructing input source of type PoolSource

   [2] Calling RootFileSequenceBase::initTheFile()

   [3] Calling StorageFactory::open()

   [4] Calling XrdFile::open()

Exception Message:

Failed to open the file 'root://xrootd-cms.infn.it//store/data/Run2011A/Cosmics/RAW/v1/000/160/960/049F6443-8E53-E011-A943-003048F117EA.root'

   Additional Info:

      [a] XrdCl::File::Open(name='root://eoscms.cern.ch//eos/cms/store/data/Run2011A/Cosmics/RAW/v1/000/160/960/049F6443-8E53-E011-A943-003048F117EA.root', flags=0x10, permissions=0660) => error '[ERROR] Server responded with an error: [3010] Unable to give access - user access restricted - not authorized identity used ; Permission denied

' (errno=3010, code=400). No additional data servers were found.

      [b] Last URL tried: root://eoscms.cern.ch:1094//eos/cms/store/data/Run2011A/Cosmics/RAW/v1/000/160/960/049F6443-8E53-E011-A943-003048F117EA.root?tried=

      [c] Problematic data server: eoscms.cern.ch:1094

      [d] Disabled source: eoscms.cern.ch:1094

      [e] Input file root://eoscms.cern.ch//eos/cms/store/data/Run2011A/Cosmics/RAW/v1/000/160/960/049F6443-8E53-E011-A943-003048F117EA.root could not be opened.

Fallback Input file root://xrootd-cms.infn.it//store/data/Run2011A/Cosmics/RAW/v1/000/160/960/049F6443-8E53-E011-A943-003048F117EA.root also could not be opened.

Original exception info is above; fallback exception info is below.

      [f] XrdCl::File::Open(name='root://xrootd-cms.infn.it//store/data/Run2011A/Cosmics/RAW/v1/000/160/960/049F6443-8E53-E011-A943-003048F117EA.root', flags=0x10, permissions=0660) => error '[ERROR] Server responded with an error: [3010] Unable to give access - user access restricted - not authorized identity used ; Permission denied

' (errno=3010, code=400). No additional data servers were found.

      [g] Last URL tried: root://eoscms.cern.ch:1094//store/data/Run2011A/Cosmics/RAW/v1/000/160/960/049F6443-8E53-E011-A943-003048F117EA.root?tried=

      [h] Problematic data server: eoscms.cern.ch:1094

      [i] Disabled source: eoscms.cern.ch:1094

----- End Fatal Exception -------------------------------------------------

</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13071/10742/summary.html

pgunnell committed:
added snippet files for tune uncertainties

Dr15Jones committed:
Make JobReport::inputFileOpened sufficiently thread-safe

Now that the mixing module runs multiple threads, we can get multiple threads each calling JobReport::inputFileOpened and trying to open files for the mixing. The code path which is called must be thread safe. Although the container used for this purpose was already thread safe, the way the item in the container was used was not safe.

davidlange6 committed:
Merge pull request #13004 from kpedro88/Phase1-QIE1

use QIEType instead of old transient qieIndex

davidlange6 committed:
Merge pull request #13083 from Dr15Jones/threadFixForJobReport

Make JobReport::inputFileOpened sufficiently thread-safe

davidlange6 committed:
Merge pull request #13059 from civanch/cleanup_warn_static_analyser2

Fixed warnings reported by the static analyzer (2nd round)

davidlange6 committed:
Merge pull request #13045 from davidlt/fix-xrootd-unused-return

[RFC] Utilities/XrdAdaptor: fix -Wunused-result errors

cmsbuild committed:
Merge pull request #13057 from lveldere/fastsim-trackval-fix

bug fix: don't run conversion track validation for fastsim

cmsbuild committed:
Merge pull request #13053 from Martin-Grunewald/MTport

Migration of modules used in PbPb HLT to do ZeroSuppression to "stream" modules in 80X

davidlange6 committed:
Merge pull request #13039 from davidlt/remove-return-createIfNeeded

Removes subsequent unused return statement in createIfNeeded

cmsbuild committed:
Merge pull request #13078 from pgunnell/TuneUncertSnippets

added snippet files for tune uncertainties

davidlange6 committed:
Merge pull request #13073 from lveldere/fastsim-cmsdriver-scenarios

fastsim : exit cmsDriver when using option --fast with fastsim-incompatible scenarios

lveldere commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: f6e92960c370c908363cbd22d01c15063741f467
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13071/10781/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13071/10781/summary.html

lveldere commented:
unexpected changes in the run1 fastsim workflow
I close until that is fixed

Commit:
Merged refs/pull/13071/head from repository cms-sw

Commit:
fastsim: fix l1reco for run1

cmsbuild commented:
Pull request #13071 was updated. @covarell, @lveldere, @cerminar, @fabozzi, @mommsen, @diguida, @bendavid, @vanbesien, @govoni, @civanch, @franzoni, @cmsbuild, @davidlange6, @vciulli, @smuzaffar, @Dr15Jones, @cvuosalo, @mdhildreth, @deguio, @slava77, @ggovi, @mmusich, @ssekmen, @boudoul, @emeschi, @thuer, @srimanob, @hengne can you please check and sign again.
